keras_contrib/applications/densenet.py

```diff
@@ -382,10 +382,10 @@ def __dense_block(x, nb_layers, nb_filter, growth_rate, bottleneck=False, dropou
     x_list = [x]
 
     for i in range(nb_layers):
-        x = __conv_block(x, growth_rate, bottleneck, dropout_rate, weight_decay)
-        x_list.append(x)
+        conv_block = __conv_block(x, growth_rate, bottleneck, dropout_rate, weight_decay)
+        x_list.append(conv_block)
 
-        x = concatenate(x_list, axis=concat_axis)
+        x = concatenate([x, conv_block], axis=concat_axis)
```

Comment:
This is the important fix that I mentioned. With it, and Deconvolution upsampling, the parameter count exactly matches the paper and trains properly. 

The previous version had fewer parameters, and did not train properly.